src/dmap_import/util_rds.py

```diff
@@ -255,9 +290,38 @@ def truncate_table(
         self.execute(sa.text(f"{truncate_query};"))
 
         # Execute VACUUM to avoid non-deterministic behavior during testing
-        with self.session.begin() as cursor:
-            cursor.execute(sa.text("END TRANSACTION;"))
-            cursor.execute(sa.text(f"VACUUM (ANALYZE) {truncat_as};"))
+        self.vaccuum_analyze(table_to_truncate)
+
+
+def copy_local_to_db(
+    local_path: str, destination_table: DeclarativeBase
+) -> None:
+    """
+    Load local file into DB using psql COPY command
+    will throw if psql command does not exit with code 0
+
+    :param local_path: path to local file that will be loaded
+    :param destination_table: SQLAlchemy table object for COPY destination
+    """
+    with gzip.open(local_path, "rt") as gzip_file:
+        local_columns = gzip_file.readline().strip().lower().split(",")
+
+    copy_command = (
+        f"\\COPY {destination_table.__table__} "
+        f"({','.join(local_columns)}) "
+        "FROM PROGRAM "
+        f"'gzip -dc {local_path}' "
+        "WITH CSV HEADER"
+    )
+
+    psql = [
+        "psql",
```

Comment:
question: I thought from @mzappitello 's experiments you could run the`COPY` command from a normal SQL connection: does that not with with `FROM PROGRAM`?

Reply:
[this](https://www.psycopg.org/docs/cursor.html#cursor.copy_expert) is the psycopg command i was using for my tests. its using the same underlying SQL Copy functionality as the command here.

i didn't run any performance tests, as my ticket was concerned with poorly formatted data rather than comparing methods for performance.

Reply:
Right, but my understanding is that the Psycopg command you were using was the same underlying Python connection. Calling out to `psql` creates a new connection to the database, that maybe we don't need?

Reply:
Accessing `psycopg` functions through `SQLAlchemy` doesn't have first class support.

Personally, I prefer calling `psql` directly to avoid the additional layers of abstraction. The cost of an additional connection that lives for a few seconds seems fair.